hudi-flink-writer/src/main/java/org/apache/hudi/schema/FilebasedSchemaProvider.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.schema;
+
+import org.apache.avro.Schema;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hudi.common.config.TypedProperties;
+import org.apache.hudi.common.fs.FSUtils;
+import org.apache.hudi.exception.HoodieIOException;
+import org.apache.hudi.util.StreamerUtil;
+
+import java.io.IOException;
+import java.util.Collections;
+
+/**
+ * A simple schema provider, that reads off files on DFS.
+ */
+public class FilebasedSchemaProvider extends SchemaProvider {
```

Comment:
> we have `TableSchemaResolver` to handle the schema

`TableSchemaResolver`  helps to read schema from data files and log files, here we need a Util to read schema from job config

Reply:
ah ok, I got confused by the class name. Should we move this to `TableSchemaResolver` then, so we have schema related stuff in one place. WDYT?

Reply:
> ah ok, I got confused by the class name. Should we move this to `TableSchemaResolver` then, so we have schema related stuff in one place. WDYT?

Yes, keep schema related stuff together makes sense. 
Maybe we can leave `TableSchemaResolver` where it is and move `SchemaProvider` stuff to `hudi-client-common` , because `TableSchemaResolver`  basically will not be modified in the future, while `SchemaProvider`  is expansion-oriented, more likely to add new child implementation, and more like client stuff.

besides, this `SchemaProvider` comes from `hudi-utilities` module, and there are still lots of utils can be reused there, we can file a new ticket to do things like this together